src/WebUI/Pods/PodsRightPanel.tsx

```diff
@@ -73,43 +110,45 @@ export class PodsRightPanel extends BaseComponent<IPodRightPanelProps, IPodsRigh
         }
 
         return (
-            <Page className="pods-right-panel-container flex flex-grow">
-                {this._getHeader()}
-                <TabBar
-                    className={"pods-right-tab-bar"}
-                    selectedTabId={this.state.selectedTab || PodsRightPanelTabsKeys.PodsDetailsKey}
-                    onSelectedTabChanged={this._onSelectedTabChanged}
-                    tabSize={TabSize.Tall}
-                >
-                    <Tab
-                        name={Resources.OverviewText}
-                        id={PodsRightPanelTabsKeys.PodsDetailsKey}
-                    />
-                    <Tab
-                        name={Resources.LogsText}
-                        id={PodsRightPanelTabsKeys.PodsLogsKey}
-                    />
-                    <Tab
-                        name={Resources.YamlText}
-                        id={PodsRightPanelTabsKeys.PodsYamlKey}
-                    />
-                </TabBar>
-                <div className="pod-details-right-content page-content page-content-top">
-                    {this._getPageContent()}
-                </div>
-            </Page>
+            this.state.pod ? (
+                <Page className="pods-right-panel-container flex flex-grow">
+                    {this._getHeader()}
+                    <TabBar
+                        className={"pods-right-tab-bar"}
+                        selectedTabId={this.state.selectedTab || PodsRightPanelTabsKeys.PodsDetailsKey}
+                        onSelectedTabChanged={this._onSelectedTabChanged}
+                        tabSize={TabSize.Tall}
+                    >
+                        <Tab
+                            name={Resources.OverviewText}
+                            id={PodsRightPanelTabsKeys.PodsDetailsKey}
+                        />
+                        <Tab
+                            name={Resources.LogsText}
+                            id={PodsRightPanelTabsKeys.PodsLogsKey}
+                        />
+                        <Tab
+                            name={Resources.YamlText}
+                            id={PodsRightPanelTabsKeys.PodsYamlKey}
+                        />
+                    </TabBar>
+                    <div className="pod-details-right-content page-content page-content-top">
+                        {this._getPageContent()}
+                    </div>
+                </Page>)
+                : <Spinner className={"flex-grow flex-center"} label={Resources.LoadingText} />
         );
     }
 
     private _getHeader(): JSX.Element | null {
-        return (
+        return this.state.pod ? (
```

Comment:
do we need these changes of pod check? as we are not going to call these untill pod is available?